Exception :
    - An exception is a problem that arises during the execution of a program. A C++ exception is a response to an exceptional circumstance that arises while a program is running,
    such as an attempt to divide by zero.

throw − A program throws an exception when a problem shows up. This is done using a throw keyword.

catch − A program catches an exception with an exception handler at the place in a program where you want to handle the problem. The catch keyword indicates the catching of an exception.

try − A try block identifies a block of code for which particular exceptions will be activated. It's followed by one or more catch blocks.



- catch (...) default exception
- 




/------------------------------------------------------------------------------/

    // Private Members ----- //
    private:
    std::string         _name;

    // Public Members ------ //
    public:

    // ---------------- Constructors & Destructors ------------- //
    ClapTrap();
    ~ClapTrap();
    ClapTrap(const ClapTrap &clap_trap);

    // ----------------- Getters & Setters ------------------------ //
  

    // ------------------- Members Public functions ---------------- //
  

    // ------------------  Operator Overload ----------------------- //
                //  ***** Assignement ********* //
    ClapTrap    &operator = (ClapTrap const &clap_trap) ;

/------------------------------------------------------------------------------/

